web/html/src/manager/visualization/hierarchy.tsx

```diff
@@ -53,140 +34,202 @@ function showFilterTab(tabIdToShow) {
   Utils.adjustSvgDimensions();
 }
 
-// util function for adding the UI to the dom and setting its callbacks
-function initUI(tree) {
-  const filterNavTab = d3.select("#visualization-filter-wrapper").append("ul").attr("class", "nav nav-tabs");
-
-  filterNavTab
-    .append("li")
-    .attr("id", "filtering-tab-selector")
-    .attr("class", "filter-tab-selector active")
-    .append("a")
-    .text(t("Filtering"))
-    .on("click", (d) => {
-      showFilterTab("filtering-tab");
-    });
-  filterNavTab
-    .append("li")
-    .attr("id", "partitioning-tab-selector")
-    .attr("class", "filter-tab-selector")
-    .append("a")
-    .text(t("Partitioning"))
-    .on("click", (d) => {
-      showFilterTab("partitioning-tab");
-    });
+// NB! This is a magic constant
```

Comment:
Here and below, this is not actually new code, almost all of it has been lifted out of the old jQuery etc code and modified to work with React.